src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs

```diff
@@ -45,12 +193,24 @@ public void WorkItemFiler_ValidatesSarifLogFileLocationArgument()
             action.Should().Throw<ArgumentNullException>();
         }
 
-        private static SarifWorkItemFiler CreateWorkItemFiler()
-            => CreateMockSarifWorkItemFiler();
+        private static SarifWorkItemFiler CreateWorkItemFiler(SarifWorkItemContext context = null)
+            => CreateMockSarifWorkItemFiler(out _, context).Object;
 
-        private static SarifWorkItemFiler CreateMockSarifWorkItemFiler()
+        private static Mock<SarifWorkItemFiler> CreateMockSarifWorkItemFiler(out Mock<FilingClient> mockClient, SarifWorkItemContext context = null)
         {
-            var mockFiler = new Mock<SarifWorkItemFiler>();
+            context = context ?? GitHubTestContext;
+
+            mockClient = new Mock<FilingClient>();
+
+            mockClient
+                .Setup(x => x.Connect(It.IsAny<string>()))
+                .CallBase();
```

Comment:
>CallBase [](start = 17, length = 8)

CallBase says to call the actual method of the type being mocked. But the type being mocked is FilingClient, and its Connect method (and its FileWorkItems method, below) are abstract. So I don't see how this works.

Reply:
Unused code! Nice catch.

---
In reply to: [389391844](https://github.com/microsoft/sarif-sdk/pull/1800#discussion_r389391844) [](ancestors = 389391844)